connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/ProtobufUtils.scala

```diff
@@ -177,47 +175,75 @@ private[sql] object ProtobufUtils extends Logging {
       .asInstanceOf[Descriptor]
   }
 
+  // TODO: Revisit to ensure that messageName is searched through all imports
   def buildDescriptor(descFilePath: String, messageName: String): Descriptor = {
-    val descriptor = parseFileDescriptor(descFilePath).getMessageTypes.asScala.find { desc =>
-      desc.getName == messageName || desc.getFullName == messageName
+    val descriptorList = parseFileDescriptor(descFilePath).map(fileDescriptor => {
+      fileDescriptor.getMessageTypes.asScala.find { desc =>
+        desc.getName == messageName || desc.getFullName == messageName
+      }
+    }).filter(f => !f.isEmpty)
+
+    if (descriptorList.isEmpty) {
+      throw QueryCompilationErrors.noProtobufMessageTypeReturnError(messageName)
     }
 
-    descriptor match {
+    descriptorList.last match {
       case Some(d) => d
       case None =>
-        throw new RuntimeException(s"Unable to locate Message '$messageName' in Descriptor")
+        throw QueryCompilationErrors.unableToLocateProtobufMessageError(messageName)
     }
   }
 
-  private def parseFileDescriptor(descFilePath: String): Descriptors.FileDescriptor = {
+  private def parseFileDescriptor(descFilePath: String): List[Descriptors.FileDescriptor] = {
```

Comment:
Rename to `parseFileDescriptorSet` (otherwise it sounds like it is parsing just one file descriptor).